collections/_posts/2025-01-18-GradientStepDenoiser.md

```diff
@@ -0,0 +1,240 @@
+
+I'll skip the details, but the important thing is that now we can devise a **convergent Plug-and-Play** algorithm, with **any neural network** $$N_\sigma$$. The algorithm reads as follows:
+
+$$ x^{n+1} = \text{Prox}_{\tau f} \circ \left( \text{Id} - \tau \lambda \nabla g_\sigma \right) (x^{n})$$
+
+which is pretty similar to the standard Forward-Backward algorithm. 
+Replacing the gradient step on $$g_\sigma$$ by its formulation as a denoiser gives:
+
+$$ x^{n+1} = \text{Prox}_{\tau f} \circ D_\sigma (x^{n})$$
+
+This algorithm is then proven to explicitly decrease the objective function:
+
+$$F(x) = f(x) + g_\sigma (x)$$
+
+which was our initial objective.
+
+**REMARK**: you might notice that it is unusual to take a proximal step on $$f$$ and a gradient step on the regularization $$g$$. This algorithm needs to be formulated as such in order to write the convergence proof, but this is not ideal. In some cases $$f$$ does not admit a closed-form solution for its proximal operator. Subsequent work from the same authors partially fixes this issue [6-7].
+
+# Experiments
+
+The authors test their denoiser along with their new algorithm on two tasks:
+- deblurring, with 10 different blur kernels (motion blur, gaussian, uniform)
+- super-resolution (with gaussians)
+
+They perform on-par with the state-of-the-art other PnP algorithms, while offering convergence guarantees.
+
+![Deblurring experiments](/collections/images/GradientStep/deblurring.jpg)
+
+The explicit objective function $$F$$ can be seen decreasing to a local minimum and reaching a fixed-point in subfigure (g).
+
+![Super-resolution experiments](/collections/images/GradientStep/superresolution.jpg)
+
+The extensive experiments of the authors show that they perform on-par with SOTA methods while using a lighter denoiser.
+
+![Quantitative results on the deblurring task](/collections/images/GradientStep/results.jpg)
+
+# Conclusion
+
+Why does this matter ?
+
+- state-of-the-art performances
+- a single network for a wide variety of image restoration problems
+- now we can confidently use deep learning models in image restoration
+- no need to select the "best" iteration, or use early-stopping: the algorithm is convergent !
+- a new framework to easily design other convergent Plug-and-Play algorithms for inverse problems
+- opens up a lot of new research directions
+
+# References
+
+[1] First-Order Methods in Optimization, Amir Beck, 2017
+
+[2] Plug-and-Play priors for model based reconstruction, Venkatakrishnan et al., 2013
+
+[3] Plug-and-Play Image Restoration With Deep Denoiser Prior, Zhang et al., 2022
+
+[4] Plug-and-Play Methods Provably Converge with Properly Trained Denoisers, Ryu et al., 2019
+
+[5] Plug-and-Play Methods Provably Converge with Properly Trained Denoisers, Terris et al., 2020
```

Comment:
The titles of both references are the same, which I suspect was a copy-paste issue :wink: